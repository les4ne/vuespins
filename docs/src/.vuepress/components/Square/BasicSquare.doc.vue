<template>
  <div>
    <Square />
  </div>
</template>

<script>
import { Square } from 'vuespins'

export default {
  name: 'app',
  components: { Square }
}
</script>
