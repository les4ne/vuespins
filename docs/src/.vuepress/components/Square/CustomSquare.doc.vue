<template>
  <div>
    <Square color="#fb503b" :size="80" :speed="5" />
  </div>
</template>

<script>
import { Square } from 'vuespins'

export default {
  name: 'app',
  components: { Square }
}
</script>
