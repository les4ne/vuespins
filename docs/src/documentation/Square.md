# Square

## **Basic**

<br>
<client-only>
<Square />
</client-only>

### Props

Basic "Square" component

| property | propType | required | default | custom |
| :------: | :------: | :------: | :-----: | :----: |
| `color`  |  string  |  false   | #282c34 |   -    |
|  `size`  |  number  |  false   |   50    |   -    |
| `speed`  |  number  |  false   |   1.2   |   -    |

### Usage

<<< @/src/.vuepress/components/Square/BasicSquare.doc.vue{3,8,10-13}

## **Custom**

<br>
<client-only>
<Square color="#fb503b" :size="80" :speed="5" />
</client-only>

### Props

Custom "Square" component

| property | propType | required | default | custom  |
| :------: | :------: | :------: | :-----: | :-----: |
| `color`  |  string  |  false   | #282c34 | #fb503b |
|  `size`  |  number  |  false   |   50    |   80    |
| `speed`  |  number  |  false   |   1.2   |    5    |

### Usage

<<< @/src/.vuepress/components/Square/CustomSquare.doc.vue{3,8,10-13}
