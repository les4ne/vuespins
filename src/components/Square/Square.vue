<template>
  <div class="vs-square" :style="cssVars"></div>
</template>

<script>
export default {
  name: 'vs-square',
  props: {
    color: {
      type: String,
      default: '#282c34'
    },
    size: {
      type: String,
      default: '50'
    },
    speed: {
      type: String,
      default: '1.2'
    }
  },
  computed: {
    cssVars() {
      return {
        '--vs-square-size': this.size + 'px',
        '--vs-square-color': this.color,
        '--vs-square-speed': this.speed + 's'
      }
    }
  }
}
</script>

<style scoped>
.vs-square {
  width: var(--vs-square-size);
  height: var(--vs-square-size);
  background-color: var(--vs-square-color);
  -webkit-animation: vs-rotateplane var(--vs-square-speed) infinite ease-in-out;
  animation: vs-rotateplane var(--vs-square-speed) infinite ease-in-out;
}

@-webkit-keyframes vs-rotateplane {
  0% {
    -webkit-transform: perspective(120px);
  }
  50% {
    -webkit-transform: perspective(120px) rotateY(180deg);
  }
  100% {
    -webkit-transform: perspective(120px) rotateY(180deg) rotateX(180deg);
  }
}

@keyframes vs-rotateplane {
  0% {
    transform: perspective(120px) rotateX(0deg) rotateY(0deg);
    -webkit-transform: perspective(120px) rotateX(0deg) rotateY(0deg);
  }
  50% {
    transform: perspective(120px) rotateX(-180.1deg) rotateY(0deg);
    -webkit-transform: perspective(120px) rotateX(-180.1deg) rotateY(0deg);
  }
  100% {
    transform: perspective(120px) rotateX(-180deg) rotateY(-179.9deg);
    -webkit-transform: perspective(120px) rotateX(-180deg) rotateY(-179.9deg);
  }
}
</style>
