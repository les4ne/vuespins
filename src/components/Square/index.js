import Square from './Square.vue'

export default Square
