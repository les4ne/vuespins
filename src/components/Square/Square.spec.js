import { shallowMount } from '@vue/test-utils'
import Square from './Square.vue'

describe('Square.vue', () => {
  it('renders correctly', () => {
    expect(shallowMount(Square).element).toMatchSnapshot()
  })
})
