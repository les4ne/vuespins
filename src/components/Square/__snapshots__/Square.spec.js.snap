// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square.vue renders correctly 1`] = `
<div
  class="vs-square"
  style="--vs-square-size: 50px; --vs-square-color: #282c34; --vs-square-speed: 1.2s;"
/>
`;
